core/sql/optimizer/RelExpr.cpp

```diff
@@ -1434,6 +1449,44 @@ void RelExpr::pushdownCoveredExpr(const ValueIdSet & outputExpr,
       computeValuesReqdForPredicates(predicatesOnParent,
                                      exprToEvalOnParent);
 
+      // -----------------------------------------------------------------
+      // Check for equality predicates that could not be pushed down.
+      // This may happen if we have an equality predicate that was not
+      // transformed into a VEG predicate. 
+      //
+      // If there are column references in the equality predicate, it
+      // may prove useful to tell the leaf nodes about it, as the equality
+      // predicate might be pushed down later during a Join to TSJ 
+      // transformation. Telling the leaf nodes about it might allow the
+      // use of an index in this case (see Scan::addIndexInfo; this is
+      // called once at the beginning of optimization, before any Join
+      // to TSJ transformations have been attempted).
+      // -----------------------------------------------------------------
+      ValueId pred;
+      for (pred = predicatesOnParent.init();
+           predicatesOnParent.next(pred);
+           predicatesOnParent.advance(pred))
+        {
+          ItemExpr * ie = pred.getItemExpr();
+          if (ie->getOperatorType() == ITM_EQUAL) // non-VEG, equality predicate
+            {
+              for (iter = firstChild; iter < lastChild; iter++)
+                {
+                  for (CollIndex i = 0; i < ie->getArity(); i++)
+                    {
+                      // If the child is covered, it might contain a column reference
+                      // that is useful for index selection. Push that down.
+                      ItemExpr * ieChildi = ie->child(i);
+                      if ((CmpCommon::getDefault(COMP_BOOL_194) == DF_ON) &&
+                          (coveredSubExprNotUsed[iter].contains(ieChildi->getValueId())))
```

Comment:
It may be more efficient to move the check on CB_194 out of the topmost FOR loop.

Reply:
Will do. Thanks.